retrievalmarket/impl/provider.go

```diff
@@ -26,7 +26,10 @@ import (
 	"github.com/filecoin-project/go-fil-markets/shared"
 )
 
-type provider struct {
+type RetrievalProviderOption func(p *Provider)
+type DealDecider func(ctx context.Context, state retrievalmarket.ProviderDealState) (bool, string, error)
+
+type Provider struct {
```

Comment:
Why did you decide to make this a public struct? I just want to make sure it's neccessary.

Reply:
It's similar to StorageMarket -- if you make it private you can't operate on (p Provider) to set a config option.  StorageProviderOption and RetrievalProviderOption both take a (Storage|Retrieval)Provider.